src/connector/src/source/kafka/enumerator.rs

```diff
@@ -200,6 +200,18 @@ impl SplitEnumerator for KafkaSplitEnumerator {
             })
             .collect();
 
+        let arc_client = self.client.clone();
+        {
+            #[cfg(not(madsim))]
+            tokio::task::spawn_blocking(move || {
```

Comment:
I noticed that in other places we call the poll method only when aws msk iam is used. Should we follow the same pattern here? https://github.com/risingwavelabs/risingwave/blob/578349a5498784e02f1a8146a59613bd51def963/src/connector/src/connector_common/connection.rs#L121

Reply:
+1 I think we should gate it, in case there's unexpected behaviour from polling. It won't affect users who don't use aws_msk_iam. Better to be defensive here. Especially when we do debugging in the future, we can be sure this code path is not executed with non aws_msk_iam users and ignore it.